ElevatorSim/src/elevatorsim/floor/FloorServer.java

```diff
@@ -56,17 +84,40 @@ public void sendElevatorRequest(ElevatorRequest request) {
 			System.out.println("FloorServer - ERROR: Could not find Request Message");
 			e.printStackTrace();
 		}
+		
+		
+		
 	}
-
+	
+	/**
+	 * Sends an Elevator Request to the Scheduler
+	 * @param request the elevator request
+	 */
+	public void sendDestinationRequest(ElevatorDestinationRequest request) {
+		try {
+			sender.send(MessagePackets.generateElevatorButtonRequest(request), InetAddress.getByName(NetworkConstants.SCHEDULER_IP), NetworkConstants.SCHEDULER_PORT);
+		} catch (UnknownHostException e) {
+			System.out.println("FloorServer - ERROR: Could not find Scheduler Host. Check Network Set Up");
+			e.printStackTrace();
+		} catch (IOException e) {
+			System.out.println("FloorServer - ERROR: Could not find Request Message");
+			e.printStackTrace();
+		}
+	}
+	
 	/**
 	 * Notifies the Scheduler that the Floor System is terminating
 	 */
 	public void sendExitRequest() {
 		try {
+			Thread.sleep(TimeConstants.initiateExitDelay);
```

Comment:
The constant seems to be for the time it takes for people to get off the elevator - we don't need a sleep here

Reply:
This is to make the floorServer wait a while after it has sent the last floorRequest message so that an elevator has time to service the request

Reply:
oh ok then